Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -1083,3 +1087,41 @@ private func qualtricsConfigData() -> QualtricsConfigData {
     .compact()
   )
 }
+
+private func shouldGoToLandingPage() -> Bool {
+  guard AppEnvironment.current.currentUser == nil else {
+    AppEnvironment.current.ubiquitousStore.hasSeenLandingPage = true
+    AppEnvironment.current.userDefaults.hasSeenLandingPage = true
+
+    return false
+  }
+
+  let userAttributes = optimizelyUserAttributes(
+    with: AppEnvironment.current.currentUser,
+    project: nil,
+    refTag: nil
+  )
+
+  let optimizelyVariant = AppEnvironment.current.optimizelyClient?
+    .variant(
+      for: OptimizelyExperiment.Key.nativeOnboarding,
+      userId: deviceIdentifier(uuid: UUID()),
+      isAdmin: AppEnvironment.current.currentUser?.isAdmin ?? false,
+      userAttributes: userAttributes
+    )
+
+  guard let variant = optimizelyVariant else {
+    return false
+  }
+
+  switch variant {
+  case .variant1, .variant2:
+    return [
+      AppEnvironment.current.ubiquitousStore.hasSeenLandingPage,
+      AppEnvironment.current.userDefaults.hasSeenLandingPage
+    ]
+    .allSatisfy(isFalse)
+  case .control:
```

Comment:
For interest sake - if you wanted to you could lose the `guard` above and change this to `case .control, nil` to cover the `nil` case.

But fine either way 👍